### PR TITLE
Add implementation and tests for replace scalar function

### PR DESCRIPTION
**Pull request by alpaylan**
Adds `replace` scalar function.

### CODE REVIEW

File: core/translate/expr.rs [hunk 1]
```diff
@@ -1550,6 +1550,57 @@ pub fn translate_expr(
                             });
                             Ok(target_register)
                         }
+                        ScalarFunc::Replace => {
+                            let args = if let Some(args) = args {
+                                if !args.len() == 3 {
+                                    crate::bail_parse_error!(
+                                        "{} function with wrong number of arguments",
```
Comment (jussisaurio) [inline core/translate/expr.rs:1557]: could we improve on sqlite's error message here and say `function {}() requires exactly 3 arguments` or similar?

File: core/translate/expr.rs [hunk 2]
```diff
@@ -1550,6 +1550,57 @@ pub fn translate_expr(
...
+                                        srf.to_string()
+                                    )
+                                }
+                                args
+                            } else {
+                                crate::bail_parse_error!(
+                                    "{} function with no arguments",
```
Comment (jussisaurio) [inline core/translate/expr.rs:1564]: can just put the same error message here as well

File: core/translate/expr.rs (hunk 1 above)
Comment (alpaylan) [reply core/translate/expr.rs]: I've changed the message with the latest commit.

File: core/vdbe/mod.rs [hunk 3]
```diff
@@ -3422,6 +3429,37 @@ fn exec_cast(value: &OwnedValue, datatype: &str) -> OwnedValue {
...
+        || matches!(replacement, OwnedValue::Null)
+    {
+        return OwnedValue::Null;
+    }
+
+    let source = exec_cast(source, "TEXT");
+    let pattern = exec_cast(pattern, "TEXT");
+    let replacement = exec_cast(replacement, "TEXT");
+
+    // If any of the casts failed, return NULL
```
Comment (jussisaurio) [inline core/vdbe/mod.rs:3449]: looks like we are panicing instead of returning null, which is probably fine, but pls change this comment

**Review by jussisaurio — APPROVED** [review]
thanks so much! pls remove/edit the stale comment and this is good to go

File: core/vdbe/mod.rs (hunk 3 above)
Comment (alpaylan) [reply core/vdbe/mod.rs]: Fixed.

**Pull request merged**